jadx-gui/src/main/java/jadx/gui/device/debugger/smali/Smali.java

```diff
@@ -660,9 +660,13 @@ private void fmtCols(InsnData insn, LineInfo line) {
 	private void formatByteCode(StringBuilder smali, byte[] bytes) {
 		int maxLen = Math.min(bytes.length, 4 * 2); // limit to 4 units
 		StringBuilder inHex = new StringBuilder();
+		int bytecode = 0;
 		for (int i = 0; i < maxLen; i++) {
-			int temp = ((bytes[i++] & 0xff) << 8) | (bytes[i] & 0xff);
-			inHex.append(String.format("%04x ", temp));
+			bytecode = bytecode << 8 | bytes[i] & 0xff;
+			if (i % 2 == 1) {
+				inHex.append(String.format("%04x ", bytecode));
+				bytecode = 0;
+			}
 		}
```

Comment:
I think this can be simplified with just changing step to 2:
```java
		for (int i = 0; i < maxLen - 1; i += 2) {
			int temp = ((bytes[i++] & 0xff) << 8) | (bytes[i] & 0xff);
			inHex.append(String.format("%04x ", temp));
		}
```
Anyway previous version must be output incorrect bytes :(
Please test before submit.

Reply:
I tested it, and it outputs the same result with the previous code.

Reply:
Haha, I also messed this time ))
No need to increment `i` inside loop:
```java
		for (int i = 0; i < maxLen - 1; i += 2) {
			int temp = ((bytes[i] & 0xff) << 8) | (bytes[i + 1] & 0xff);
			inHex.append(String.format("%04x ", temp));
		}
```

Reply:
Okay.